staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/marshal_test.go

```diff
@@ -117,6 +290,114 @@ func TestJSONSchemaPropsOrArrayUnmarshalJSON(t *testing.T) {
+	cases := []struct {
+		name          string
+		input         string
+		expected      JSONSchemaPropsOrArrayHolder
+		expectTypeErr bool
+	}{
+		{
+			name:     "empty parent",
+			input:    "\xA0",
+			expected: JSONSchemaPropsOrArrayHolder{},
+		},
+		{
+			name:     "props zero value",
+			input:    "\xA1\x44val1\xA0",
+			expected: JSONSchemaPropsOrArrayHolder{JSPoA: JSONSchemaPropsOrArray{Schema: &JSONSchemaProps{}}},
+		},
+		{
+			name:     "props",
+			input:    "\xA1\x44val1\xA1\x44type\x46string",
+			expected: JSONSchemaPropsOrArrayHolder{JSPoA: JSONSchemaPropsOrArray{Schema: &JSONSchemaProps{Type: "string"}}},
+		},
+		{
+			name:     "array props zero value",
+			input:    "\xA1\x44val1\x81\xA0",
+			expected: JSONSchemaPropsOrArrayHolder{JSPoA: JSONSchemaPropsOrArray{JSONSchemas: []JSONSchemaProps{{}}}},
+		},
+		{
+			name:     "array props zero value and props",
+			input:    "\xA1\x44val1\x82\xA0\xA1\x44type\x46string",
+			expected: JSONSchemaPropsOrArrayHolder{JSPoA: JSONSchemaPropsOrArray{JSONSchemas: []JSONSchemaProps{{}, {Type: "string"}}}},
+		},
+		{
+			name:     "omitempty props zero value",
+			input:    "\xA1\x44val2\xA0",
+			expected: JSONSchemaPropsOrArrayHolder{JSPoAOmitEmpty: &JSONSchemaPropsOrArray{Schema: &JSONSchemaProps{}}},
+		},
+		{
+			name:     "omitempty props",
+			input:    "\xA1\x44val2\xA1\x44type\x46string",
+			expected: JSONSchemaPropsOrArrayHolder{JSPoAOmitEmpty: &JSONSchemaPropsOrArray{Schema: &JSONSchemaProps{Type: "string"}}},
+		},
+		{
+			name:     "omitempty array props zero value",
+			input:    "\xA1\x44val2\x81\xA0",
+			expected: JSONSchemaPropsOrArrayHolder{JSPoAOmitEmpty: &JSONSchemaPropsOrArray{JSONSchemas: []JSONSchemaProps{{}}}},
+		},
+		{
+			name:     "omitempty array props zero value and props",
+			input:    "\xA1\x44val2\x82\xA0\xA1\x44type\x46string",
+			expected: JSONSchemaPropsOrArrayHolder{JSPoAOmitEmpty: &JSONSchemaPropsOrArray{JSONSchemas: []JSONSchemaProps{{}, {Type: "string"}}}},
+		},
+		{
+			name:  "wrong type",
+			input: "\xA1\x44val1\x18\x2A", // {"val1":42}
+		},
+		{
+			name:  "null",
+			input: "\xF6", // null
+		},
+	}
+
+	for _, c := range cases {
+		t.Run(c.name, func(t *testing.T) {
+			var result JSONSchemaPropsOrArrayHolder
+			err := cbor.Unmarshal([]byte(c.input), &result)
+			if c.expectTypeErr {
+				if err == nil || !strings.Contains(err.Error(), "JSON schema or array of JSON schema expected") {
+					t.Fatalf("Expected error 'JSON schema or array of JSON schema expected', got %v", err)
+				}
+			} else {
+				if (err != nil) != c.expectTypeErr {
+					t.Errorf("Unmarshal error = %v, wantErr %v", err, c.expectTypeErr)
+				}
+			}
+
+			if !reflect.DeepEqual(result, c.expected) {
+				t.Errorf("Failed to unmarshal input %X", c.input)
+				t.Logf(cmp.Diff(c.expected, result))
+			}
+		})
+	}
+
+	t.Run("null", func(t *testing.T) {
+		var actual JSONSchemaPropsOrArray
+		err := cbor.Unmarshal([]byte("\xF6"), &actual)
```

Comment:
Same thing for the CBOR unmarshal null cases, does this actually call UnmarshalCBOR?